### PR TITLE
FIX: Related items order not same in editor and ninjs output. [SDESK-5083]

### PR DESCRIPTION
**Pull request by GyanP**
Introduced order in related items.
requires : https://github.com/superdesk/superdesk-client-core/pull/3371

### CODE REVIEW

File: superdesk/publish/formatters/ninjs_formatter.py [hunk 1]
```diff
@@ -301,14 +300,13 @@ def _get_associations(self, article, subscriber):
...
-        associations = OrderedDict()
+        associations = {}
         extra_items = {}
         media = {}
         content_profile = None
         archive_service = superdesk.get_resource_service('archive')
-        sorted_associations = OrderedDict(sorted(article.get(ASSOCIATIONS).items() or {}))
 
-        for key, item in sorted_associations.items():
+        for key, item in article.get(ASSOCIATIONS).items() or {}:
```
Comment (ride90) [inline superdesk/publish/formatters/ninjs_formatter.py]: you can type `article.get(ASSOCIATIONS, {}).items()`

File: superdesk/data_updates/00022_20200309-121947_archive.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from superdesk.commands.data_updates import DataUpdate
+
+
+class DataUpdate(DataUpdate):
+
+    resource = 'archive'
+
+    def forwards(self, mongodb_collection, mongodb_database):
+        related = list(get_resource_service('vocabularies').get(req=None, lookup={'field_type': 'related_content'}))
+        archive_service = get_resource_service('archive')
```
Comment (ride90) [inline superdesk/data_updates/00022_20200309-121947_archive.py]: why do we need it if we use `mongodb_database[resource]` ?

File: superdesk/data_updates/00022_20200309-121947_archive.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        for resource in ('archive', 'published'):
+            service = get_resource_service(resource)
+            for item in mongodb_database[resource].find({'associations': {'$gt': {}}}):
```
Comment (ride90) [inline superdesk/data_updates/00022_20200309-121947_archive.py]: IMO this filter condition is weird. 
Wouldn't it be more explicit to use something like that
`{ associations: { $exists: true, $ne: {} }`  (not tested, just guessed)

File: superdesk/data_updates/00022_20200309-121947_archive.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from superdesk.vocabularies import is_related_content
+from superdesk.commands.data_updates import DataUpdate
+
+
+class DataUpdate(DataUpdate):
+
+    resource = 'archive'
+
+    def forwards(self, mongodb_collection, mongodb_database):
+        related = list(get_resource_service('vocabularies').get(req=None, lookup={'field_type': 'related_content'}))
```
Comment (ride90) [inline superdesk/data_updates/00022_20200309-121947_archive.py]: this is redundant, it's done in `is_related_content` if `related_content` arg is not provided.

File: superdesk/data_updates/00022_20200309-121947_archive.py (hunk 2 above)
Comment (GyanP) [reply superdesk/data_updates/00022_20200309-121947_archive.py]: i forgot to remove.

File: superdesk/data_updates/00022_20200309-121947_archive.py (hunk 4 above)
Comment (GyanP) [reply superdesk/data_updates/00022_20200309-121947_archive.py]: good catch.I didn't looked at `is_related_content` definition.

**Pull request merged**